scylla/src/transport/topology.rs

```diff
@@ -91,9 +93,18 @@ impl TopologyReader {
     pub async fn read_topology_info(&mut self) -> Result<TopologyInfo, QueryError> {
         let mut result = self.fetch_topology_info().await;
 
+        // shuffle known_peers to iterate through them in random order later
+        self.known_peers.shuffle(&mut thread_rng());
```

Comment:
Suggestion for an optimization: if we managed to successfully fetch the topology info from the `control_connection_address` node, then we can return immediately and skip shuffling `known_peers`.

Reply:
Done